lib/chef/provider/aws_iam_role.rb

```diff
@@ -0,0 +1,37 @@
+require 'chef/provisioning/aws_driver/aws_provider'
+
+class Chef::Provider::AwsIamRole < Chef::Provisioning::AWSDriver::AWSProvider
+  provides :aws_iam_role
+
+  def action_create
+    iam_role = super
+  end
+
+  protected
+
+  def create_aws_object
+    iam = new_resource.driver.iam_resource
+
+    converge_by "create IAM role #{new_resource.name}" do
+      iam.create_role({
+        path: new_resource.path,
+        role_name: new_resource.name,
+        assume_role_policy_document: new_resource.assume_role_policy_document
+      })
+    end
+  end
+
+  def update_aws_object(iam_role)
```

Comment:
It looks like you can update the inline policy document with http://docs.aws.amazon.com/sdkforruby/api/Aws/IAM/RolePolicy.html#put-instance_method